File: FournilFront/src/app/models/product.ts
import { Format } from './format';
import { ImagesProduct } from './images-product';
import {PhotoProduct} from './photo-product';
import { ProductInCart } from './product-in-cart';

export class Product{

    private _id : number;
	private _productName : String;
	private _productSummary : String;
	private _productSpecification : String;
	private _productPerKgPrice : number;
	private _productComposition : String;
    private _limitStockProduct : number;
    private _photoProductSet: PhotoProduct[];
    private _formatSet:Format[];
    private _quantity:number;

	constructor(data:any) {
		this._id = data.id;
		this._productName = data.productName;
		this._productSummary = data.productSummary;
		this._productSpecification = data.productSpecification;
		this._productPerKgPrice = data.productPerKgPrice;
		this._productComposition = data.productComposition;
		this._limitStockProduct = data.limitStockProduct;
		this._photoProductSet = data.photoProductSet;
        this._formatSet = data.formatSet;
        this._quantity=data.quantity
	}  

    


    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter productName
     * @return {String}
     */
	public get productName(): String {
		return this._productName;
	}

    /**
     * Getter productSummary
     * @return {String}
     */
	public get productSummary(): String {
		return this._productSummary;
	}

    /**
     * Getter productSpecification
     * @return {String}
     */
	public get productSpecification(): String {
		return this._productSpecification;
	}

    /**
     * Getter productPerKgPrice
     * @return {number}
     */
	public get productPerKgPrice(): number {
		return this._productPerKgPrice;
	}

    /**
     * Getter productComposition
     * @return {String}
     */
	public get productComposition(): String {
		return this._productComposition;
	}

    /**
     * Getter limitStockProduct
     * @return {number}
     */
	public get limitStockProduct(): number {
		return this._limitStockProduct;
	}

    /**
     * Getter photoProductSet
     * @return {PhotoProduct[]}
     */
	public get photoProductSet(): PhotoProduct[] {
		return this._photoProductSet;
	}

    /**
     * Getter formatSet
     * @return {Format[]}
     */
	public get formatSet(): Format[] {
		return this._formatSet;
	}


    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter productName
     * @param {String} value
     */
	public set productName(value: String) {
		this._productName = value;
	}

    /**
     * Setter productSummary
     * @param {String} value
     */
	public set productSummary(value: String) {
		this._productSummary = value;
	}

    /**
     * Setter productSpecification
     * @param {String} value
     */
	public set productSpecification(value: String) {
		this._productSpecification = value;
	}

    /**
     * Setter productPerKgPrice
     * @param {number} value
     */
	public set productPerKgPrice(value: number) {
		this._productPerKgPrice = value;
	}

    /**
     * Setter productComposition
     * @param {String} value
     */
	public set productComposition(value: String) {
		this._productComposition = value;
	}

    /**
     * Setter limitStockProduct
     * @param {number} value
     */
	public set limitStockProduct(value: number) {
		this._limitStockProduct = value;
	}

    /**
     * Setter photoProductSet
     * @param {PhotoProduct[]} value
     */
	public set photoProductSet(value: PhotoProduct[]) {
		this._photoProductSet = value;
	}

    /**
     * Setter formatSet
     * @param {Format[]} value
     */
	public set formatSet(value: Format[]) {
		this._formatSet = value;
	}


    /**
     * Getter quantity
     * @return {number}
     */
	public get quantity(): number {
		return this._quantity;
	}

    /**
     * Setter quantity
     * @param {number} value
     */
	public set quantity(value: number) {
		this._quantity = value;
	}


   

}

File: FournilBack/src/main/java/fr/fournil/bakery/model/entities/Product.java
package fr.fournil.bakery.model.entities;

import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.validation.constraints.Min;

import com.fasterxml.jackson.annotation.*;


@Entity
@Table(name = "productSet")
//@Inheritance(strategy=InheritanceType.JOINED)
//@MappedSuperclass
public class Product {
	 private static final long serialVersionUID = 1L;
	 @Id
	 @GeneratedValue(strategy = GenerationType.IDENTITY)
	 protected Long id;
	
	private String productName;
	@Column(columnDefinition = "TEXT")
	private String productSummary;
	
	@Column(columnDefinition = "TEXT")
	private String productComposition;
	
	@Column(columnDefinition = "TEXT")
	private String productSpecification;

	
   @OneToMany(mappedBy = "productIdName", cascade = {CascadeType.PERSIST}, fetch = FetchType.EAGER)
   //@JsonManagedReference
   //@JsonIgnore
	private Set<PhotoProduct> photoProductSet = new HashSet<PhotoProduct>();
	
   @ManyToMany(mappedBy = "productList", cascade = { CascadeType.PERSIST }, fetch = FetchType.EAGER)
  // @JsonManagedReference
   private Set<Format> formatSet = new HashSet<Format>();
	
 
   @OneToOne(mappedBy = "product", cascade = {CascadeType.PERSIST}, fetch = FetchType.EAGER)
   @JsonIgnore
   private ProductInCart productInCart;
   
    private double productPerKgPrice;
	
    @Min(0)
	private Integer limitStockProduct;
	
	

	public Product() {
		super();
	}

	

	public Product(Long id, String productName, String productSummary, String productComposition,
			String productSpecification, Set<PhotoProduct> photoProductSet, Set<Format> formatSet,
			ProductInCart productInCart, double productPerKgPrice, @Min(0) Integer limitStockProduct) {
		super();
		this.id = id;
		this.productName = productName;
		this.productSummary = productSummary;
		this.productComposition = productComposition;
		this.productSpecification = productSpecification;
		this.photoProductSet = photoProductSet;
		this.formatSet = formatSet;
		this.productInCart = productInCart;
		this.productPerKgPrice = productPerKgPrice;
		this.limitStockProduct = limitStockProduct;
	}


	public Product(String productName, String productSummary, String productComposition, String productSpecification,
			Set<PhotoProduct> photoProductSet, Set<Format> formatSet, ProductInCart productInCart,
			double productPerKgPrice, @Min(0) Integer limitStockProduct) {
		super();
		this.productName = productName;
		this.productSummary = productSummary;
		this.productComposition = productComposition;
		this.productSpecification = productSpecification;
		this.photoProductSet = photoProductSet;
		this.formatSet = formatSet;
		this.productInCart = productInCart;
		this.productPerKgPrice = productPerKgPrice;
		this.limitStockProduct = limitStockProduct;
	}



	public Long getId() {
		return id;
	}



	public void setId(Long id) {
		this.id = id;
	}



	public String getProductName() {
		return productName;
	}



	public void setProductName(String productName) {
		this.productName = productName;
	}



	public String getProductSummary() {
		return productSummary;
	}



	public void setProductSummary(String productSummary) {
		this.productSummary = productSummary;
	}



	public String getProductComposition() {
		return productComposition;
	}



	public void setProductComposition(String productComposition) {
		this.productComposition = productComposition;
	}



	public String getProductSpecification() {
		return productSpecification;
	}



	public void setProductSpecification(String productSpecification) {
		this.productSpecification = productSpecification;
	}



	public Set<PhotoProduct> getPhotoProductSet() {
		return photoProductSet;
	}



	public void setPhotoProductSet(Set<PhotoProduct> photoProductSet) {
		this.photoProductSet = photoProductSet;
	}



	public Set<Format> getFormatSet() {
		return formatSet;
	}



	public void setFormatSet(Set<Format> formatSet) {
		this.formatSet = formatSet;
	}



	public ProductInCart getProductInCart() {
		return productInCart;
	}



	public void setProductInCart(ProductInCart productInCart) {
		this.productInCart = productInCart;
	}



	public double getProductPerKgPrice() {
		return productPerKgPrice;
	}



	public void setProductPerKgPrice(double productPerKgPrice) {
		this.productPerKgPrice = productPerKgPrice;
	}



	public Integer getLimitStockProduct() {
		return limitStockProduct;
	}



	public void setLimitStockProduct(Integer limitStockProduct) {
		this.limitStockProduct = limitStockProduct;
	}



	public static long getSerialversionuid() {
		return serialVersionUID;
	}



	@Override
    public int hashCode() {
        return Objects.hash(id, productName);
    }



	@Override
	public String toString() {
		return "Product [id=" + id + ", productName=" + productName + ", productSummary=" + productSummary
				+ ", productComposition=" + productComposition + ", productSpecification=" + productSpecification
				 + ", productPerKgPrice=" + productPerKgPrice + ", limitStockProduct="
				+ limitStockProduct + "]";
	}

	


	
	



	

	
}
